### ISSUE TITLE
Panic in RegionAssembly reproducibly

### ISSUE THREAD

**Issue by frehberg** [georust/geo#1127]
Occasionally (1:8) I observe a panic condition. I managed to print out the data of RegionAssembly causing the panic, and using this data to create a regression test (see PR)

Code with regresssion test assembling 3668 segments/lines  https://github.com/georust/geo/pull/1126

The panic occures in the the check  in file `assembly.rs`
```
            debug_assert!(num_segments % 2 == 0, "assembly segments must be eulierian");
```

The execution path leading to this point of the RegionAssembly  occures in the context of `bop.sweep()` in the following code

```
impl<T: GeoFloat> BooleanOps for MultiPolygon<T> {
    type Scalar = T;

    fn boolean_op(&self, other: &Self, op: OpType) -> MultiPolygon<Self::Scalar> {
        let spec = BoolOp::from(op);
        let mut bop = Proc::new(spec, self.coords_count() + other.coords_count());
        bop.add_multi_polygon(self, 0);
        bop.add_multi_polygon(other, 1);
        bop.sweep()
    }
```   
Appreciating any hint why this crash occures and how to avoid the panic crash

![Screenshot from 2023-12-21 11-30-58](https://github.com/georust/geo/assets/4435830/bbc30c64-1e99-489c-9f0b-0842e5dd95cd)

**Comment by urschrei** [comment]
#1102, #1103, and #1104 may be related here.

**Comment by frehberg** [comment]
is the code of RegionAssembly cashing due to a bug in the algorithm, or due to bad input parameters? In case of the latter, how to prevent this?

**Comment by urschrei** [comment]
It's crashing due to a bug, and a fix is unlikely in the short term.

**Comment by frehberg** [comment]
hopefully this  test  https://github.com/georust/geo/pull/1126 can help you to reproduce the issue and fix the code

**Comment by michaelkirk** [comment]
I believe this is a dupe of https://github.com/georust/geo/issues/913

That issue is still open - but let's try to focus any future discussion on this bug there. Note there may be other bugs with the boolean operations - this one in particular is about the "assembly segments must be eulierian", which I believe is caused by a collapse of floating point precision.

Feel free to re-open if I'm mistaken and this represents a different bug.

**Comment by michaelkirk** [comment]
I just merged #1234 which replaces our BoolOps implementation with one backed by the i_overlay crate. This should resolve the issue you're seeing. Let us know if it recurs. You can use it now if you use the unreleased `geo` crate, otherwise we expect it to be part of the upcoming v0.29.0 release.
